dialer.go

```diff
@@ -209,24 +270,26 @@ func (c *Dialer) Publisher(opts ...publisher.Option) (*publisher.Publisher, erro
 func (c *Dialer) connectState() {
 	defer close(c.connCh)
 	defer close(c.closedCh)
-	defer close(c.unreadyCh)
 	defer c.cancelFunc()
 	defer c.logger.Printf("[DEBUG] connection closed")
 
 	i := 0
 	l := len(c.amqpUrls)
 
 	c.logger.Printf("[DEBUG] connection unready")
+
 	var connErr error = amqp.ErrClosed
+
+	c.notifyUnready(connErr)
```

Comment:
there is no need for connErr anymore.  Please do 
```suggestion
	c.notifyUnready(amqp.ErrClosed)
```

Reply:
+